codec.proto

```diff
@@ -10,3 +12,22 @@ package weave;
 message Metadata {
   uint32 schema = 1;
 }
+
+// CommitInfo is basically a replica of tendermint commit info
```

Comment:
Yes, this is all abci stuff, can you use a similar approach in #747 to pull in the ValidatorUpdate struct?

Probably best to merge this PR first, then rebase #747 and do those API changes